addons/mail_plugin/models/res_partner.py

```diff
@@ -1,8 +1,78 @@
 # -*- coding: utf-8 -*-
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
-    iap_enrich_info = fields.Text('IAP Enrich Info', help='Stores additional info retrieved from IAP in JSON')
+    iap_enrich_info = fields.Text('IAP Enrich Info', help='IAP response stored as a JSON string',
+                                  compute='_compute_iap_enrich_info')
+
+    search_domain = fields.Char('Search Domain / Email',
+                                compute='_compute_search_domain')
+
+    @api.model
+    def create(self, vals):
+        partner = super(ResPartner, self).create(vals)
+
+        if vals.get('iap_enrich_info') or vals.get('search_domain'):
+            # Not done with inverse method so we do not need to search
+            # for existing <res.partner.iap>
+            self.env['res.partner.iap'].sudo().create({
+                'partner_id': partner.id,
+                'iap_enrich_info': vals.get('iap_enrich_info'),
+                'search_domain': vals.get('search_domain'),
+            })
+
+        return partner
+
+    def write(self, vals):
+        super(ResPartner, self).write(vals)
+
+        if 'iap_enrich_info' in vals or 'search_domain' in vals:
+            # Not done with inverse method so we do need to search
+            # for existing <res.partner.iap> only once
+            partner_iaps = self._get_partner_iaps()
+            missing_partners = self
+            for partner_iap in partner_iaps:
+                if 'iap_enrich_info' in vals:
+                    partner_iap.iap_enrich_info = vals['iap_enrich_info']
+                if 'search_domain' in vals:
+                    partner_iap.search_domain = vals['search_domain']
+
+                missing_partners -= partner_iap.partner_id
+
+            if missing_partners:
+                # Create new <res.partner.iap> for missing records
+                self.env['res.partner.iap'].sudo().create([
+                    {
+                        'partner_id': partner.id,
+                        'iap_enrich_info': vals.get('iap_enrich_info'),
+                        'search_domain': vals.get('search_domain'),
+                    } for partner in missing_partners
+                ])
+
+    def _compute_iap_enrich_info(self):
```

Comment:
Can't we group the two compute ?

+ guidelines: compute should be before CRUD

Reply:
I just had the idea before seeing your second review :)